sandbox/Example.JetStream.PullConsumer/Program.cs

```diff
@@ -19,7 +19,7 @@
 
 var js = new NatsJSContext(nats);
 
-var consumer = await js.CreateConsumerAsync(new ConsumerCreateRequest { StreamName = "s1", Config = new ConsumerConfiguration { Name = "c1", DurableName = "c1", AckPolicy = ConsumerConfigurationAckPolicy.@explicit } });
+var consumer = await js.CreateConsumerAsync("s1", new ConsumerConfiguration { Name = "c1", DurableName = "c1", AckPolicy = ConsumerConfigurationAckPolicy.@explicit });
```

Comment:
Framework guidelines dictate that enum names should be Pascal cased -- then you could avoid the ugly at escape on explicit.

Reply:
Unfortunately I had an issue with the JSON serializer not using the attributed name then had to make it same case as the JSON property to make it work.

Reply:
For S.T.Json, the way I do this [net8] is:

```csharp
builder.Services
   .AddJsonOptions(
        options =>
        {
            options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter<PassportIssuingSettings>(JsonNamingPolicy.CamelCase));
            options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
            
            options.JsonSerializerOptions.TypeInfoResolver = PassportJsonSerializationContext.Default;
 ```

Reply:
PassportJsonSerializationContext is my source generator context for my structures. I'd imagine you guys are using that code path too?

Reply:
Here's the partial context I declared: 

```csharp
[JsonSerializable(typeof(Passport))]
[JsonSerializable(typeof(PassportIssuingSettings))]
[JsonSerializable(typeof(Visa))]
[JsonSerializable(typeof(ValidationProblemDetails))]
[JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
public partial class PassportJsonSerializationContext : JsonSerializerContext
{
}
```
This is for AOT ready serialization etc.

Reply:
Tbh, some of my declarations are probably redundant - anything in the AddJsonOptions setup can probably be applied to the context instead. I went through an evolution making things work from net6 to net8 :/

Reply:
Yes we have a context, that attribute solution looks promising. Do you still have to set the option?

Reply:
I think the option is only for when you don't use a context. I started out that way, then added a context later but never took away the option. I _assume_ the attributes on the context are enough :)

Reply:
Oh, so you can pass in options director to the context ctor too:

```csharp
_jsonSerializationContext = new PassportJsonSerializationContext(
    new JsonSerializerOptions
    {
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
        Converters = {  }
    });
 ```
 
 Then you don't need the entry point AddJsonOptions fluff.

Reply:
hmmm. it doesn't seem to work (see example https://github.com/mtmk/example-json-enum)

Need to implement a convertor I think.